ISPN-12044 CLI performance tool

tristantarrant opened the pull request:
https://issues.redhat.com/browse/ISPN-12044

tristantarrant commented:
This is now ready

karesti commented on cli/cli-client/src/main/java/org/infinispan/cli/benchmark/HttpBenchmark.java:
```diff
@@ -0,0 +1,116 @@
+   public void setup() {
+      RestURI uri = RestURI.create(this.uri);
+      RestClientConfigurationBuilder builder = uri.toConfigurationBuilder();
+      cm = RestClient.forConfiguration(builder.build());
```
cm variable name related to cacheManager but the RestClient returns a restClient. I would rename this to avoid confusion with the HotrodBenchmark

karesti commented on cli/cli-client/src/main/java/org/infinispan/cli/benchmark/HttpBenchmark.java:
```diff
@@ -0,0 +1,116 @@
+      RestClientConfigurationBuilder builder = uri.toConfigurationBuilder();
+      cm = RestClient.forConfiguration(builder.build());
+      cache = cm.cache(cacheName);
+      if (uncheckedAwait(cache.exists()).getStatus() == 404) {
```
same here, it's not a cache but the restCacheClient

karesti reviewed (changes requested):
small comments, I guess @oraNod may want to grab a look to docs as well?

oraNod commented on cli/cli-client/src/main/resources/help/benchmark.adoc:
```diff
@@ -0,0 +1,71 @@
+----
+benchmark - runs a performance benchmark against a cache.
+
+The benchmark can be performed both for HTTP and for Hot Rod: the mode is determined by the protocol of the supplied URI.
```
@tristantarrant I'm curious about "the mode is determined by the protocol of the supplied URI." How does the protocol determine the mode? Are there some benchmark modes that do not apply for certain protocols?

oraNod commented on cli/cli-client/src/main/resources/help/benchmark.adoc:
```diff
@@ -0,0 +1,71 @@
+The size of the test key set. Defaults to 1000.
+
+*--verbosity*=['SILENT', 'NORMAL', 'EXTRA']::
+How verbose the output should be. One of SILENT, NORMAL, EXTRA. Defaults to NORMAL.
```
There seems to be no difference in the output when I set different verbosity levels.

oraNod commented on cli/cli-client/src/main/resources/help/benchmark.adoc:
```diff
@@ -0,0 +1,71 @@
+The name of the cache to use for the benchmark. Defaults to 'benchmark'.
+
+**--key-size*='num'::
+The size of the key. Defaults to 16 bytes.
```
The size unit for the value is byte? "Sets the size, in bytes, of the key.

oraNod commented on cli/cli-client/src/main/resources/help/benchmark.adoc:
```diff
@@ -0,0 +1,71 @@
+The number of measurement iterations to perform. Defaults to 5.
+
+*--time*='time'::
+How long each iteration should take. Defaults to 10s.
```
The time unit is seconds? "Sets the amount of time, in seconds, that each iteration takes."

tristantarrant commented:
@oraNod @karesti I believe I've addressed the issues

oraNod commented:
@tristantarrant hey, I pushed a commit with suggested doc edits. feel free to revert, squash, whatever. also I noticed you removed the fork option from the command but it was still in the help text so I removed it there. cheers.

tristantarrant commented:
@oraNod I've squashed your commit into mine
@karesti I've addressed most comments, aside from those related to the BenchmarkOutputFormat which I've explained

oraNod reviewed (approved):
LGTM
